backend/src/governance.cairo

```diff
@@ -0,0 +1,215 @@
+                authorAddress: author,
+                status: Status::TRADING,
+                marketAddress: market, //TODO : marketAddress should not be a parameter but the address of a market contract deployed for this proposal
+                oracleAddress: oracle, //TODO : same 
+                tradingPhaseDurationInSeconds: tradingDuration.try_into().unwrap(),
+                adoptionPhaseDurationInSeconds: adoptionDuration.try_into().unwrap(),
+            };
+            //First let's add our proposal id to our array of proposal ids
+            let mut allProposals = self.getProposalsIds();
+            allProposals.append(proposalCreated.id);
+            self.proposalsIds.write(allProposals);
+
+            // Then add the proposal to our mapping of proposal by id
+            self.proposalById.write(proposalCreated.id, proposalCreated);
+            
+            // Finally add the proposal id to the mapping of proposals ids by author address
+            let mut proposalsIdFromAuthor = self.getAuthorProposalsIds(author);
+            proposalsIdFromAuthor.append(id.into());
+            self.proposalsByAuthor.write(author, proposalsIdFromAuthor);
+        }
+
+        // Return all proposal ids
+        fn getProposalsIds(self: @ContractState) -> Array<felt252> {
+            self.proposalsIds.read()
+        }
+
+        // Return proposal ids from an author
+        fn getAuthorProposalsIds(self: @ContractState, author: ContractAddress) -> Array<felt252> {
+            self.proposalsByAuthor.read(author)
+        }
+
+        // Return proposals from an author
+        fn getAuthorProposals(self: @ContractState, author: ContractAddress) -> Array<Proposal> {
+            let mut ids = self.getAuthorProposalsIds(author);
+            let mut arrayOfProposals: Array<Proposal> = array![];
+            while let Option::Some(element) = ids.pop_front() {
+                arrayOfProposals.append(self.getProposalById(element));
+            };
+            arrayOfProposals
+        }
+
+        // Return the proposal corresponding to its id
+        fn getProposalById(self: @ContractState, id: felt252) -> Proposal {
+            self.proposalById.read(id)
+        }
+
+        // Return all proposals
+        fn getProposals(self: @ContractState) -> Array<Proposal> {
+            let mut ids = self.getProposalsIds();
+            let mut arrayOfProposals: Array<Proposal> = array![];
+            while let Option::Some(element) = ids.pop_front() {
+                arrayOfProposals.append(self.getProposalById(element));
+            };
+            arrayOfProposals
+        }
+        
+        fn resolveTradingPhase(ref self: ContractState) {
+            //TODO
+        }
+
+        fn resolveAdoptionPhase(ref self: ContractState) {
+            //TODO
+        }
+
+        
+    }
+}
+
+// This block of code is used to store Array<felt252> in the Storage.
+impl StoreFelt252Array of starknet::Store<Array<felt252>> {
```

Comment:
It's a utility trait implementation that allows us to store array of felt252 in our contract